apps/settings/js/modules/addon_manager.js

```diff
@@ -63,6 +63,10 @@ define(function(require) {
       if (this._isAddon(app)) {
         if (type === 'install' && !this._alreadyExists(app)) {
           this._addons.push(App(app));
+          // If there's an app based filter, apply it.
+          if (this._filter) {
+            this._applyFilter();
```

Comment:
I would suggest to do the filtering outside of the addon manager, which give us more flexibility on having different types of filters.